### PR TITLE
Initial merge to ONS repo

### PR DESCRIPTION
**Pull request by KamenDimitrov97**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+FROM python:3.10-slim
+
+RUN apt-get update && apt-get install --no-install-recommends -y git \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN pip install "poetry==1.1.12"
+RUN pip install urllib3==1.26.15
```
Comment (lindenmckenzie) [inline Dockerfile:8]: This really needs to be abstracted to a config file.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+[tool.poetry.dev-dependencies]
+pytest = "^6.2.5"
+flake8 = "^4.0.1"
+pylint = "^2.15.4"
+pylint-quotes = "^0.2.3"
+pytest-cov = "^3.0.0"
+pytest-xdist = "^2.5.0"
+pytest-sugar = "^0.9.4"
+black = "^22.10.0"
+berlin = "^0.3.0"
```
Comment (lindenmckenzie) [inline pyproject.toml:75]: why is berlin a dev dependency?

**Review by lindenmckenzie — COMMENTED** [review]
Some initial comments, pending getting it running.

**Review by lindenmckenzie — COMMENTED** [review]
Some general comments prior to full review.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    "pytest-sugar>=0.9.4,<0.10",
+    "black>=22.10.0,<22.11"
+]
+
+[tool.black]
+exclude = "tests/"
+target_version = ['py38']
+
+[tool.ruff]
+# Ignore E501 - some lines can't be smalled than 90
```
[the remark this answers is not in the document]
Comment (KamenDimitrov97) [reply pyproject.toml:43]: smalled :D I must have been in a hurry

File: app/views/berlin.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    logger.info("Search schema")
+    return jsonify({}), 200
+
+
+@berlin_blueprint.route("/berlin/search", methods=["GET"])
+def berlin_search():
+    q = request.args.get("q")
+    state = request.args.get("state")
+    limit = request.args.get("limit", type=int)
+    lev_distance = request.args.get("lev_distance", 2, type=int)
```
[the remark this answers is not in the document]
Comment (KamenDimitrov97) [reply app/views/berlin.py:41]: Only the "q" query is required and if not supplied returns a 500. The others are optional.

**Pull request closed without merging**